lib/puppet/ssl/host.rb

```diff
@@ -237,10 +233,24 @@ def validate_certificate_with_key
     end
   end
 
+  # Search for an existing CSR for this host either cached on
+  # disk or stored by the CA. Returns nil if no request exists.
+  # @return [Puppet::SSL::CertificateRequest, nil]
+  def certificate_request
+    unless @certificate_request
+      if csr = load_certificate_request_from_file
+        @certificate_request = csr
+      elsif Puppet::SSL::Host.ca_location == :remote && csr = download_csr_from_ca
```

Comment:
I don't think we should put an assignment in the second clause of a test expression.

Reply:
That's fair, I'll break this up.